stellar-contract-env-common/src/env_val.rs

```diff
@@ -168,7 +168,9 @@ impl<E: Env> TryFrom<EnvVal<E, RawVal>> for i64 {
         if ev.val.is_positive_i64() {
             Ok(unsafe { ev.val.unchecked_as_positive_i64() })
         } else if Object::val_is_obj_type(ev.val, ScObjectType::I64) {
-            Ok(ev.env.obj_to_i64(ev.val))
+            Ok(ev
+                .env
+                .obj_to_i64(unsafe { Object::unchecked_from_val(ev.val) }))
         } else {
             Err(())
         }
```

Comment:
This code doesn't need to do a check and then unchecked from val, because you can use the TryInto trait impl that exists, and then rely on the host checking I64 specifically. This keeps at least one branch out of the guest WASM code.

```rust
if ev.val.is_positive_i64() {
    Ok(unsafe { ev.val.unchecked_as_positive_i64() })
} else {
    let ev: EnvVal<E, Object> = ev.try_into()?;
    Ok(ev.env.obj_to_i64(ev.val))
}
```

Reply:
Thanks, updated.